### PR TITLE
[snapshot-controller] Switch to distroless

### PR DESCRIPTION
**Pull request by duckhawk**
## Description

Switch snapshot-controller to distroless

## Why do we need it, and what problem does it solve?

We need to reduce the number of attack vectors

## What is the expected result?

More secure snapshot-controller module

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: snapshot-controller
type: chore
summary: Switch _snapshot-controller_ module to distroless.
```

### CODE REVIEW

File: modules/045-snapshot-controller/images/snapshot-validation-webhook/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
+    add: /snapshot-validation-webhook
+    to: /snapshot-validation-webhook
+    before: setup
+docker:
+  ENTRYPOINT: ["/snapshot-validation-webhook"]
+
+---
+artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
+from: {{ $.Images.BASE_GOLANG_20_ALPINE }}
```
Comment (RomanenkoDenys) [inline modules/045-snapshot-controller/images/snapshot-validation-webhook/werf.inc.yaml:14]:
```suggestion
from: {{ $.Images.BASE_GOLANG_22_ALPINE_DEV }}
```

File: modules/045-snapshot-controller/images/snapshot-validation-webhook/werf.inc.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+mount:
+  - fromPath: ~/go-pkg-cache
+    to: /go/pkg
+shell:
+  beforeInstall:
+    - apk add --no-cache ca-certificates make git
```
Comment (RomanenkoDenys) [inline modules/045-snapshot-controller/images/snapshot-validation-webhook/werf.inc.yaml:20]:
```suggestion
```

File: modules/045-snapshot-controller/images/snapshot-controller/werf.inc.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
+    add: /snapshot-controller
+    to: /snapshot-controller
+    before: setup
+docker:
+  ENTRYPOINT: ["/snapshot-controller"]
+
+---
+artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
+from: {{ $.Images.BASE_GOLANG_20_ALPINE }}
```
Comment (RomanenkoDenys) [inline modules/045-snapshot-controller/images/snapshot-controller/werf.inc.yaml:14]:
```suggestion
from: {{ $.Images.BASE_GOLANG_22_ALPINE_DEV }}
```

File: modules/045-snapshot-controller/images/snapshot-controller/werf.inc.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+mount:
+  - fromPath: ~/go-pkg-cache
+    to: /go/pkg
+shell:
+  beforeInstall:
+    - apk add --no-cache ca-certificates make git
```
Comment (RomanenkoDenys) [inline modules/045-snapshot-controller/images/snapshot-controller/werf.inc.yaml:20]:
```suggestion
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Review by RomanenkoDenys — APPROVED** [review]

**Review by AleksZimin — APPROVED** [review]

**Pull request merged**